third_party/jemalloc.BUILD

```diff
@@ -31,7 +31,23 @@ JEMALLOC_BUILD_COMMAND = """
   export LTOFLAGS="$$([ "$$(uname)" = "Linux" ] && echo "-flto=thin")" # todo: on next clang toolchain upgrade, check if it's fixed and we can re-enable thinlto on mac
   export EXTRA_CFLAGS="$${LTOFLAGS}"
   export EXTRA_CXXFLAGS="-stdlib=libc++ $${EXTRA_CFLAGS}"
-  export LDFLAGS="$${LTOFLAGS} $$([ "$$(uname)" = "Linux" ] && echo " -fuse-ld=lld")"
+
+  LDFLAGS="$${LTOFLAGS}"
+
+  case "$$(uname)" in
+    Linux)
+      LDFLAGS="$${LDFLAGS} -fuse-ld=lld"
+      ;;
+    Darwin)
+      LDFLAGS="$${LDFLAGS} -mlinker-version=400"
+      ;;
+  esac
+
+  export LDFLAGS
+
+  # we need to include CFLAGS in CPPFLAGS so that the --sysroot flag makes it in
+  export CPPFLAGS="$${CFLAGS}"
```

Comment:
So great that the jemalloc build system has `EXTRA_CXXFLAGS` above but `CPPFLAGS` here.

Actually, this script has `export CFLAGS=$(CC_FLAGS)`/`export CXXFLAGS=$(CC_FLAGS)` just a few lines above this, and we don't seem to be making additional changes to `CFLAGS` that would affect this...is this just necessary because the script should be using `export CXXFLAGS=$$(CC_FLAGS)`?  Or is it because the script should have been setting `CPPFLAGS` instead of `CXXFLAGS` in the first place?